dbms/src/Common/TiFlashSecurity.h

```diff
@@ -121,9 +145,9 @@ struct TiFlashSecurityConfig
         }
     }
 
-
     bool checkGrpcContext(const grpc::ServerContext * grpc_context) const
     {
+        std::unique_lock lock(mu);
```

Comment:
I think this mutex is unnecessary.

Reply:
It is necessary. When reloading config, other threads may call `checkGrpcContext`, there is a data race.